src/AWS.Deploy.Orchestration/OptionSettingHandler.cs

```diff
@@ -333,5 +349,85 @@ public bool IsSummaryDisplayable(Recommendation recommendation, OptionSettingIte
 
             return true;
         }
+
+        /// <summary>
+        /// Checks whether the option setting item has been modified by the user. If it has been modified, then it will hold a non-default value
+        /// </summary>
+        /// <returns>true if the option setting item has been modified or false otherwise</returns>
+        public bool IsOptionSettingModified(Recommendation recommendation, OptionSettingItem optionSetting)
+        {
+            // If the option setting is not displayable, that means its dependencies are not satisfied and it does not play any role in the deployment.
+            // We do not need to evaluate whether it has been modified or not.
+            if (!IsOptionSettingDisplayable(recommendation, optionSetting))
+            {
+                return false;
+            }
+
+            if (optionSetting.Type.Equals(OptionSettingValueType.List))
+            {
+                var currentSet = GetOptionSettingValue<SortedSet<string>>(recommendation, optionSetting);
+                var defaultSet = GetOptionSettingDefaultValue<SortedSet<string>>(recommendation, optionSetting);
+
+                if (defaultSet == null)
+                    return currentSet?.Count > 0;
```

Comment:
i believe we could get an exception here. what happens if currentSet is null? currentSet?.Count would be null. does `null > 0` throw an error?